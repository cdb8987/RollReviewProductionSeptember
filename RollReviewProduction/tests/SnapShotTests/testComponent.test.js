import React from 'react'
import {View, Text} from 'react-native'
import renderer, {act} from 'react-test-renderer';
import TestComponent from './TestComponent.js'
import testComponentChild from './testComponentChild.js'

test('Counter Button Increments by 1', ()=>{
    //Arrange
    const tree = renderer.create(<TestComponent/>)
    const testInstance = tree.root

    const countDisplay = testInstance.findByProps({testID: 'countDisplay'})
    const incrementButton = testInstance.findByProps({testID: 'incrementCount'})
    //check initial state
    expect(countDisplay.props.children).toEqual(0);
    
    // Act
    act(()=>{incrementButton.props.onPress()})
    
    expect(countDisplay.props.children).toEqual(1);
    act(()=>{incrementButton.props.onPress()})
    expect(countDisplay.props.children).toEqual(2);

    expect(tree).toMatchSnapshot()
    
    
    

    //Assert
})
