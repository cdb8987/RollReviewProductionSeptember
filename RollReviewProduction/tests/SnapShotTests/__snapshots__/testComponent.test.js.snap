// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Counter Button Increments by 1 1`] = `
<View>
  <View
    accessibilityState={
      {
        "busy": undefined,
        "checked": undefined,
        "disabled": undefined,
        "expanded": undefined,
        "selected": undefined,
      }
    }
    accessibilityValue={
      {
        "max": undefined,
        "min": undefined,
        "now": undefined,
        "text": undefined,
      }
    }
    accessible={true}
    collapsable={false}
    focusable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      {
        "opacity": 1,
      }
    }
    testID="incrementCount"
  >
    <Text>
      incrementCount
    </Text>
  </View>
  <testComponentChild />
  <Text
    testID="countDisplay"
  >
    2
  </Text>
</View>
`;
